sormas-backend/src/main/java/de/symeda/sormas/backend/report/WeeklyReportService.java

```diff
@@ -168,12 +169,11 @@ public WeeklyReport getByEpiWeekAndUser(EpiWeek epiWeek, User user) {
 	public Predicate createUserFilter(CriteriaBuilder cb, CriteriaQuery cq, From<WeeklyReport, WeeklyReport> from) {
 
 		User currentUser = getCurrentUser();
-		if (currentUser == null) {
-			return null;
-		}
-
 		// National users can access all reports in the system
-		if (currentUser.hasAnyUserRole(UserRole.NATIONAL_USER, UserRole.NATIONAL_CLINICIAN, UserRole.NATIONAL_OBSERVER, UserRole.REST_USER)) {
+		final JurisdictionLevel jurisdictionLevel = currentUser.getJurisdictionLevel();
+		if (currentUser == null
+				|| (jurisdictionLevel == JurisdictionLevel.NATION && !currentUser.hasAnyUserRole(UserRole.POE_NATIONAL_USER))
```

Comment:
I'd still suggest to check for POE in general and not the specific POE_NATIONAL... user role.

Reply:
In which way was this resolved? I don't see any code changes here; Martin's suggestion makes sense to me because it shouldn't be possible to have two user roles from different jurisdictions in the first place

Reply:
it's a mistake this was also for aggregate report service and changed only there